client-coroutines/src/main/kotlin/io/provenance/client/coroutines/PbCoroutinesClient.kt

```diff
@@ -122,31 +129,67 @@ open class PbCoroutinesClient(
     suspend fun broadcastTx(
         baseReq: BaseReq,
         gasEstimate: GasEstimate,
-        mode: ServiceOuterClass.BroadcastMode = ServiceOuterClass.BroadcastMode.BROADCAST_MODE_SYNC
-    ): ServiceOuterClass.BroadcastTxResponse {
-        return cosmosService.broadcastTx(
-            ServiceOuterClass.BroadcastTxRequest
-                .newBuilder()
-                .setTxBytes(buildTx(baseReq, gasEstimate).toByteString())
-                .setMode(mode)
-                .build()
-        )
+        mode: BroadcastMode = BroadcastMode.BROADCAST_MODE_SYNC
+    ): BroadcastTxResponse {
+        return buildTx(baseReq, gasEstimate)
+            .emulateBlockMode(mode, baseReq.body.timeoutHeight) {
+               cosmosService.broadcastTx(it)
+            }
     }
 
     suspend fun estimateAndBroadcastTx(
         txBody: TxBody,
         signers: List<BaseReqSigner>,
-        mode: ServiceOuterClass.BroadcastMode = ServiceOuterClass.BroadcastMode.BROADCAST_MODE_SYNC,
+        mode: BroadcastMode = BroadcastMode.BROADCAST_MODE_SYNC,
         gasAdjustment: Double? = null,
         feeGranter: String? = null,
         feePayer: String? = null,
-    ): ServiceOuterClass.BroadcastTxResponse = baseRequest(
+    ): BroadcastTxResponse = baseRequest(
         txBody = txBody,
         signers = signers,
         gasAdjustment = gasAdjustment,
         feeGranter = feeGranter,
         feePayer = feePayer,
     ).let { baseReq -> broadcastTx(baseReq, estimateTx(baseReq), mode) }
+
+    private suspend fun TxOuterClass.TxRaw.emulateBlockMode(
+        mode: BroadcastMode,
+        providedTimeoutHeight: Long,
+        handler: suspend (BroadcastTxRequest) -> BroadcastTxResponse
+    ): BroadcastTxResponse {
+        val (actualMode, simulateBlock) = if (mode == BroadcastMode.BROADCAST_MODE_BLOCK) {
```

Comment:
the magic happens in here, converting the mode sent to the chain if in BLOCK mode and simulating that behavior via polling instead